### PR TITLE
Parse model value to string in valueBinding()

### LINKED ISSUE #84: Make re-binding a prop smarter
Imagine the following scenario:

```ts
// A simple component that has a value applied as an attribute
const someComponentTemplate = (props) => 
  html`<button data-component="some-component" value=${props.value}>${props.label}</button>`;

// The component also has a binding where we can modify the value.
const SomeComponent = defineComponent({
   name: 'some-component',
   props: {
    value: propType.number.source({ type: 'attr' }),  // ← Note how we state that this prop is a number
  },
  setup({ refs, props }) {
    return [
       bind(refs.self, {
         attr: {
           value: computed(() => props.value), // ← But when applying this binding we get an error as soon as we modify the value!
         }
      }),
    ]
  }
})
```
I would expect this to work but it gives me the following error in the console:

image

The workaround here would be to first parse the value back to a string like so:

```ts
bind(refs.self, {
  attr: {
    value: computed(() => props.value.toString())
  }
})
```

But I would assume that we could come up with a smarter approach here where the type is automatically inferred or something! 🤓

### PR DESCRIPTION
**Pull request by jspolancor**
- Update valueBinding() to parse the model value to string when calling checkInitialBindingState()
- Adds tests for valueBinding()

### CODE REVIEW

File: src/lib/bindings/dom/valueBinding.test.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { bind, computed, defineComponent, propType } from '../../..';
+
+describe('styleBinding', () => {
```
Comment (ThaNarie) [inline src/lib/bindings/dom/valueBinding.test.ts]:
```suggestion
describe('valueBinding', () => {
```

File: src/lib/bindings/dom/valueBinding.test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  it('should apply the binding', () => {
```
Comment (ThaNarie) [inline src/lib/bindings/dom/valueBinding.test.ts]:
```suggestion
  it('should work with non-string values', () => {
```

**Review by ThaNarie — APPROVED** [review]
Approved after applying the suggestions :)

File: src/lib/bindings/dom/valueBinding.test.ts (hunk 1 above)
Comment (jspolancor) [reply src/lib/bindings/dom/valueBinding.test.ts]: Done!

File: src/lib/bindings/dom/valueBinding.test.ts (hunk 2 above)
Comment (jspolancor) [reply src/lib/bindings/dom/valueBinding.test.ts]: Done!

**Pull request merged**